src/python/pants/jvm/resolve/lockfile_metadata.py

```diff
@@ -111,12 +99,12 @@ def _from_json_dict(
         return JVMLockfileMetadataV1(requirements)
 
     def _header_dict(self) -> dict[Any, Any]:
-        out = super()._header_dict()
-
-        out["generated_with_requirements"] = (
-            sorted(self.requirements) if self.requirements is not None else None
-        )
-        return out
+        return {
```

Comment:
We had a discussion about this pattern back when implementing Python lockfiles, that if we were only changing one or two properties, it was better to amend a dictionary rather than construct a new one. This pattern is repeated over there.

Reply:
I don't necessarily have a preference, but the inconsistency feels weird at this point :)

Reply:
Our style guide says to prefer unpacking and expressions over mutation: https://www.pantsbuild.org/docs/style-guide#prefer-merging-collections-through-unpacking. We use this pattern abundantly throughout the codebase

Reply:
OK, I can't find the previous discussion where we decided on the pattern I copied over from the Python implementation, so happy to go with this version. 

Is it worth changing the version `python/lockfile_metadata` to match?

Reply:
Sure, I can update it in followup changes, such as pending the other thread in this PR or the followup PRs I'm working on. I'd rather not burn a tree just for that change.